pyffi/object_models/xml/struct_.py

```diff
@@ -251,22 +274,20 @@ def __init__(self, template = None, argument = None, parent = None):
         # this list is used for instance by qskope to display the structure
         # in a tree view
         self._items = []
+
         # initialize attributes
+        # _attribute_list is a list of StructAttributes, including unions of the same name
         for attr in self._attribute_list:
             # skip attributes with dupiclate names
             # (for this to work properly, duplicates must have the same
             # type, template, argument, arr1, and arr2)
-            if attr.name in names:
-                continue
-            names.add(attr.name)
-
+            # if attr.name in names:
+            #     continue
+            # names.add(attr.name)
             # things that can only be determined at runtime (rt_xxx)
-            rt_type = attr.type_ if attr.type_ != type(None) \
-                      else template
-            rt_template = attr.template if attr.template != type(None) \
-                          else template
-            rt_arg = attr.arg if isinstance(attr.arg, (int, type(None))) \
-                     else getattr(self, attr.arg)
+            rt_type = attr.type_ if attr.type_ != type(None) else template
+            rt_template = attr.template if attr.template != type(None) else template
+            rt_arg = attr.arg
```

Comment:
With arg as an expression (for vertex desc) I'm not sure the rt_arg logic is still needed. But I'm also fairly sure the new code is not correct. Maybe evaluate expression instead?